content/articles/sql-injection-made-familiar/index.md

```diff
@@ -0,0 +1,218 @@
+- The form has some fundamental security, for example, sanitizing the username field.
+- This implies our above code can't be utilized to sidestep the login. To get around that, we can rather take advantage of the secret word field.
+- The outline below shows the steps that you should follow.
+
+#### Step 1
+Clone the source code in your local computer and run it with the `xammp` server.
+
+#### Step 2
+Enter log-in info :
+- username = 'xyzxyz'
+- password = 'xyz') OR 1=1 -- ]
+![hack-activity](app1.png)
+- click on submit
+
+#### Step 3
+Access Granted -> redirected to Dashboard
+
+If a user provides the following:
+
+- username = `xyzxyz`
+- password = `xyz') OR 1 = 1 -- ]`
+
+The user will be redirected to the dashboard upon clicking submit button.
+
+  **Let's take a closer look at the generated statements:**
+
+```sql
+SELECT * FROM members WHERE username = ‘xyzxyzx’ AND password = md5(‘xyz’) OR 1 = 1 — ]’);
+```
+
+![generated-statements](stmt.png)
+
+### Other SQL infusion assault types
+SQL Injections can do more damage than just bypassing the login calculations. A part of the assaults incorporate:
+- Erasing information
+- Updating information
+- Embeddings information
+- Executing orders on the worker that can download and introduce malignant projects like Trojans
+- Sending out significant information, for example, Mastercard subtleties, email, and passwords to the assailant's far-off worker.
+- Getting client login subtleties and so on.
+
+The above list isn't thorough; it simply gives you a thought of what SQL Injection is capable of.
+
+### Automation devices for SQL infusion
+In the above model, we utilized manual assault methods dependent on our huge information on SQL. There are mechanized apparatuses that can assist you with playing out the assaults all the more proficiently and inside the briefest conceivable time. 
+
+These apparatuses incorporate:
+- `Sonarqube` – https://www.sonarqube.org/
+- `sqlsus` – http://sqlsus.sourceforge.net/
+
+### Forestall SQL infusion assaults
+An association can take on the accompanying strategy to ensure itself against SQL Injection assaults:
+1. Client info ought to never be trusted – It should consistently be disinfected before it is utilized in unique SQL explanations.
+2. Put away methodology – these can exemplify the SQL explanations and treat all contributions as boundaries.
+3. Arranged explanations – arranged proclamations to work by making the SQL articulation first then, at that point regarding all submitted client information as boundaries. This has no impact on the linguistic structure of the SQL articulation.
+4. Customary articulations – these can be utilized to distinguish possible hurtful code and eliminate it before executing the SQL explanations.
+5. Information base association client access rights – just vital access rights ought to be given to accounts used to interface with the data set. This can assist with lessening what the SQL explanations can perform on the worker.

```

Comment:
```suggestion
5. Information base association client access rights – only vital access rights ought to be given to accounts used to interface with the data set. This can assist with lessening what the SQL explanations can perform on the worker.
```